pkg/testutil/leak.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2013 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testutil
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"runtime"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+)
+
+/*
+CheckLeakedGoroutine verifies tests do not leave any leaky
+goroutines. It returns true when there are goroutines still
+running(leaking) after all tests.
+
+	import "github.com/coreos/etcd/pkg/testutil"
+
+	var atLeastGo15 bool = false
+
+	func init() {
+		atLeastGo15 = testutil.GetAtLeastGo15()
+	}
+
+	func TestMain(m *testing.M) {
+		v := m.Run()
+		if v == 0 && testutil.CheckLeakedGoroutine() {
+			os.Exit(1)
+		}
+		os.Exit(v)
+	}
+
+	func TestSample(t *testing.T) {
+		defer testutil.CheckLeakedGoroutineAfterTest(t, atLeastGo15)
+		...
+	}
+
+*/
+func CheckLeakedGoroutine() bool {
```

Comment:
I think it has to be exposed because `AfterTest` has to be called explicitly for each test, inside the package we are testing. 

`TestMain` doesn't help to do some kind of global operation for each test. It just runs and lets you do operation around the whole testings.